contrib/learn_resource_weights.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python3
+
```

Comment:
This should be documented with at least some of the text from PR description.

Reply:
I've added some documentation (run the script with `--help`). The PR description already has a full paragraph about this script.